deepspeed/runtime/zero/config.py

```diff
@@ -171,10 +171,16 @@ def _initialize(self, zero_config_dict):
             ZERO_OPTIMIZATION_PARAM_PERSISTENCE_THRESHOLD,
             ZERO_OPTIMIZATION_PARAM_PERSISTENCE_THRESHOLD_DEFAULT)
 
-        self.gather_fp16_weights_on_model_save = get_scalar_param(
-            zero_config_dict,
-            ZERO_OPTIMIZATION_GATHER_FP16_WEIGHTS_ON_MODEL_SAVE,
-            ZERO_OPTIMIZATION_GATHER_FP16_WEIGHTS_ON_MODEL_SAVE_DEFAULT)
+        # config key has been renamed to use "16bit" instead of "fp16." falling back
+        # to old config name in order to preserve backwards compatibility
+        self.gather_16bit_weights_on_model_save = ZERO_OPTIMIZATION_GATHER_16BIT_WEIGHTS_ON_MODEL_SAVE_DEFAULT
+        for key in [
+                ZERO_OPTIMIZATION_GATHER_16BIT_WEIGHTS_ON_MODEL_SAVE,
+                ZERO_OPTIMIZATION_GATHER_FP16_WEIGHTS_ON_MODEL_SAVE
+        ]:
+            if key in zero_config_dict:
+                self.gather_16bit_weights_on_model_save = zero_config_dict[key]
+                break
```

Comment:
Unless I missed it, I think this change didn't preserve backward compatibility for the original: `gather_fp16_weights_on_model_save` config entry. And given that DS's config checker simply ignores entries it doesn't know, now the users with the old config that still has `gather_fp16_weights_on_model_save` will have a broken behavior.

Thanks.

Reply:
hey, i believe this specific piece of code will work. added a test to confirm (which i should have done to begin with) https://github.com/microsoft/DeepSpeed/pull/1453/commits/e04dc6a24f9d587f9d45cbd3b6873022951f49ee

Reply:
you're absolutely right! I did miss it! Thank you very much for adding a test, Justin.

Reply:
sure!